migration/migration_pkg_test.go

```diff
@@ -168,3 +168,32 @@ func TestChangeTrieNodeEncoding(t *testing.T) {
 		return nil
 	}))
 }
+
+func TestCalculateBlockCommitments(t *testing.T) {
+	testdb := pebble.NewMemTest()
+	t.Cleanup(func() {
+		require.NoError(t, testdb.Close())
+	})
+	chain := blockchain.New(testdb, utils.MAINNET, utils.NewNopZapLogger())
+	client, closeFn := feeder.NewTestClient(utils.MAINNET)
+	t.Cleanup(closeFn)
+	gw := adaptfeeder.New(client)
+
+	for i := uint64(0); i < 3; i++ {
+		b, err := gw.BlockByNumber(context.Background(), i)
+		require.NoError(t, err)
+		su, err := gw.StateUpdate(context.Background(), i)
+		require.NoError(t, err)
+		require.NoError(t, chain.Store(b, &core.BlockCommitments{}, su, nil))
+	}
+
+	require.NoError(t, testdb.Update(func(txn db.Transaction) error {
+		return calculateBlockCommitments(txn, utils.MAINNET)
+	}))
+
+	for i := uint64(0); i < 3; i++ {
+		b, err := chain.BlockCommitmentsByNumber(i)
+		require.NoError(t, err)
+		assert.NotNil(t, b.TransactionCommitment)
```

Comment:
Is there a reason why you only check `TransactionCommitment` and not `EventCommitment`?

Reply:
See https://github.com/NethermindEth/juno/pull/931#discussion_r1269063435

Reply:
If we test for the content of the TransactionCommitment then we would be able to tell whether we stored the right commitment to the the right block number. Right now, these tests are only testing for the existence of a Commitment and I think `calculateCommitments` is racey even with mutex because `block` can change in the goroutine by the time it is evaluated.